redash/tasks/queries.py

```diff
@@ -278,20 +278,30 @@ class QueryExecutionError(Exception):
     pass
 
 
+def _resolve_user(user_id, is_api_key):
+    if user_id is not None:
+        if is_api_key:
+            api_key = user_id
+            q = models.Query.query.filter(models.Query.api_key == api_key).one()
```

Comment:
Let's try to avoid using `query.filter` outside of `redash.models`. This way we can easily see what's the public surface of a model.

Reply:
161d06a2